src/backend/distributed/sql/udfs/citus_shard_indexes_on_worker/11.0-1.sql

```diff
@@ -6,7 +6,6 @@ CREATE OR REPLACE FUNCTION pg_catalog.citus_shard_indexes_on_worker(
      OUT shard_name name)
  RETURNS SETOF record
  LANGUAGE plpgsql
- SET citus.hide_shards_from_app_name_prefixes = ''
```

Comment:
this was a GUC we added on citus 11.0.1, and replaced with another name on 11.0.2 via https://github.com/citusdata/citus/pull/5920

With this in place, create extension fails with on PG 15:
```
create extension citus ;
ERROR:  invalid configuration parameter name "citus.hide_shards_from_app_name_prefixes"
DETAIL:  "citus" is a reserved prefix.
```

Seems fine to remove this?

Reply:
Should this be backported?

Reply:
Ah, this is a tricky problem.

As far as I can see, we CANNOT backport this. I mean, even if we backport, the users who created the extension before the patch release would hit the issue. 11.0-2 does not access this guc anyway.

Instead, the simplest seems that can re-define the GUC, see the changes in shared_library_init.c.

I think we should be careful when we use any GUCs in the UDF files going forward. We won't be able to drop those GUCs.

This is probably something that Postgres is too strict about with PG 15, we can ask the community to relax about that as well.

Reply:
Makes sense, but I guess we can still have a migration file that removes these lines in the next release. Simply because they don't do anything.

Reply:
Actually if we make sure it's changed in Citus 11.1 with a migration, then I don't think we have to keep it in shared_library_init.c. Because anyone upgrading to PG15 will first need to upgrade to 11.1 on PG14, thus removing this app name prefix. And removing it from the original 11.0 definition would be enough to not hit this issue for new installs either.

Reply:
Makes sense, I agree that it'd be enough to not hit the issue if we remove the GUC from the original file. 

However, (a) I don't remember we ever change those files after they are released (b) adding a GUC definition has almost no overhead for us (c) could there be any other edge cases that we cannot see now?

Given issues/bugs in upgrades are pretty costly, I have a slight inclination to keep as in this last commit. But, if you think strong enough that we should not do that, I can follow your suggestion.

Reply:
I'm fine with keeping the GUC definition and not changing the past migration, because we indeed do that almost never. But I think we should at least have a new migration that removes usage of the GUC. Simply to clean it up, because it doesn't do anything.